config/config.go

```diff
@@ -655,6 +655,20 @@ func (c *GlobalConfig) isZero() bool {
 type RuntimeConfig struct {
 	// The Go garbage collection target percentage.
 	GoGC int `yaml:"gogc,omitempty"`
+
+	// Below are guidelines for adding a new field:
+	//
+	// Confirm that adding it here, rather than via a flag, is the best approach.
```

Comment:
This is misleading.

There is no "best approach". The defining design pattern we use for "flag vs config" is if we want the configuration item to be static (at startup only) or dynamic (in configuration).

Any flag that we want to be dynamic needs to have a configuration file entry. So the question if "flag vs config" is up to the user needs.

Reply:
What I meant is "use config if you want it to be dynamic / decide if you want it to be dynamic or not".

I'll remove it, unless you have a suggestion.

Reply:
I reworded, twyt